spec/ruby_terraform/command_line/boolean_option_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'spec_helper'
+require_relative '../../../lib/ruby_terraform/command_line/boolean_option'
+
+module RubyTerraform
+  module CommandLine
+    describe BooleanOption do
+      subject(:command_line_option) { described_class.new(opt_key, value, options) }
+
+      let(:opt_key) { :snake_key }
+      let(:value) { true }
+      let(:options) { {} }
+      let(:sub) { instance_double(Lino::SubcommandBuilder) }
```

Comment:
Rather than mocking `Lino::SubcommandBuilder` I think it would be better to assert against the resulting command string. `Lino` is an implementation detail and it's also cheap to execute so better to make the tests a little more black box.

Reply:
I'm generally more a fan of state based tests than mock based tests as a rule.